src/libstd/sys/redox/fs.rs

```diff
@@ -447,7 +447,10 @@ pub fn stat(p: &Path) -> io::Result<FileAttr> {
 }
 
 pub fn lstat(p: &Path) -> io::Result<FileAttr> {
-    stat(p)
+    let fd = cvt(syscall::open(p.to_str().unwrap(),
+                               syscall::O_CLOEXEC | syscall::O_STAT | syscall::O_NOFOLLOW))?;
```

Comment:
False. O_STAT allows a file handle to be opened on a file *without read permission*, if the user has the ability to read the directory. This file handle can only be used for things like `fstat`.